From: Matthias Schmitz <matthias@sigxcpu.org>
Date: Tue, 22 Apr 2014 22:49:27 +0200
Subject: Using only one word as user-agent.

* The whitspace in the user-agent string breaks the escaping and fires
  an invalid wget call during autoconf. So autoconf always returns "no"
  when two or more uris are given in env.target.
  To reproduce the problem add "-x" to the #!/bin/bash and remove "-q" from
  $wget_opts. Then run the plugin with munin-run and watch the output:
  ...
+ wget -q -O /dev/null
http://munin-monitoring.org,http://munin-monitoring.org
+ for uri in '$urls'
+ wget --spider http://munin-monitoring.org --user-agent '"Munin' -
'http_loadtime"' --no-cache --page-requisites
Spider mode enabled. Check if remote file exists.
--2014-04-22 22:54:58--  http://munin-monitoring.org/
Resolving munin-monitoring.org (munin-monitoring.org)... 87.238.48.91,
2a02:c0:1018::fa57
Connecting to munin-monitoring.org
(munin-monitoring.org)|87.238.48.91|:80... connected.
HTTP request sent, awaiting response... 200 Ok
Length: unspecified [text/html]
Remote file exists and could contain further links,
but recursion is disabled -- not retrieving.

index.html.1: No such file or directory
Removing index.html.1.
unlink: No such file or directory

Spider mode enabled. Check if remote file exists.
--2014-04-22 22:54:58--  http://-/
Resolving - (-)... failed: Name or service not known.
wget: unable to resolve host address '-'
Spider mode enabled. Check if remote file exists.
--2014-04-22 22:54:58--  http://http_loadtime%22/
Resolving http_loadtime" (http_loadtime")... failed: Name or service not
known.
wget: unable to resolve host address 'http_loadtime"'
+ '[' 4 '!=' 0 ']'
+ echo 'no (Cannot run wget against "http://munin-monitoring.org")'
no (Cannot run wget against "http://munin-monitoring.org")
+ exit 0
---
 plugins/node.d/http_loadtime.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/node.d/http_loadtime.in b/plugins/node.d/http_loadtime.in
index 8001e20..5abfd20 100644
--- a/plugins/node.d/http_loadtime.in
+++ b/plugins/node.d/http_loadtime.in
@@ -42,7 +42,7 @@ target=${target:-"http://localhost/"}
 requisites=${requisites:-"false"}
 
 urls=`echo $target | tr "," "\n"`
-wget_opt="--user-agent \"Munin - http_loadtime\" --no-cache -q"
+wget_opt="--user-agent \"Munin_-_http_loadtime\" --no-cache -q"
 if [ "$requisites" == "true" ]; then
   wget_opt="$wget_opt --page-requisites"
 fi
